### DISCUSSION TITLE
How to do ConnectAndUpdateOrCreate?

### DISCUSSION THREAD

**Discussion by shimondoodkin** [prisma/prisma#21029] [Q&A]
### Question

How to do ConnectAndUpdateOrCreate?

line nested upserts and connections

**Comment by ludralph** [comment]
Hi @shimondoodkin 👋 

Thank you for raising this question.

If I understand correctly, you want to perform:
1. A `connect` operation
2. An `Update or Create` operation otherwise known as an `Upsert` operation

To perform `connect`, you can see this [section](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#connect) of the documentation.

To perform an `upsert` operation, you can see this [section](https://www.prisma.io/docs/concepts/components/prisma-client/crud#update-or-create-records) of the documentation.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by ludralph** [reply]
Hey @shimondoodkin,

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by shimondoodkin** [reply]
i want a future like
```
const createPost = await prisma.post.create({
  data: {
    title: 'How to create a compiler',
    content: '...',
    author: {
      connect: {
        id: 9,
      },
    },
    tags: {
      ConnectAndUpdateOrCreate: {
        where: {
          name: 'computing',
        },
        create: {
          name: 'computing',
          info: 2
        },
        update: {
          info: 2
        },
      },
    },
  },
})

```

currently i did it in steps manualy

it is impossible first to connect, i have to search for it first then if i found it to do update then to make a connect string

other wise if not found i need to construct a create string.
i want then to be connected by integer index not by search value

**Reply by ludralph** [reply]
Hi @shimondoodkin 

In that case, can you open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) and document your use case so that our team can take a look?
